source/contracts/Controller.sol

```diff
@@ -67,22 +75,27 @@ contract Controller is IController {
      * Registry for lookups [whitelisted augur contracts and dev mode can use it]
      */
 
-    function setValue(bytes32 _key, address _value) public onlyWhitelistedCallers returns(bool) {
-        registry[_key] = _value;
+    function registerContract(bytes32 _key, address _address, bytes20 _commitHash, bytes32 _fileHash) public onlyOwnerCaller returns(bool) {
+        registry[_key] = ContractDetails(_key, _address, _commitHash, _fileHash);
         return true;
     }
 
-    function removeValue(bytes32 _key) public onlyWhitelistedCallers returns(bool) {
-        registry[_key] = address(0);
+    function getContractDetails(bytes32 _key) public view returns (address, bytes20, bytes32) {
+        ContractDetails storage _details = registry[_key];
+        return (_details.contractAddress, _details.commitHash, _details.fileHash);
+    }
+
+    function unregisterContract(bytes32 _key) public onlyWhitelistedCallers returns(bool) {
```

Comment:
Should this also be `onlyOwnerCaller` now?

Reply:
Yes, will fix.